winlogbeat/config/config.go

```diff
@@ -27,7 +28,33 @@ type Validator interface {
 
 // Settings is the root of the Winlogbeat configuration data hierarchy.
 type Settings struct {
-	Winlogbeat WinlogbeatConfig
+	Winlogbeat WinlogbeatConfig       `config:"winlogbeat"`
+	All        map[string]interface{} `config:",inline"`
+}
+
+// Validate validates the Settings data and returns an error describing
+// all problems or nil if there are none.
+func (s Settings) Validate() error {
+	validKeys := []string{"winlogbeat", "output", "shipper", "logging"}
```

Comment:
That looks like something we should move to libbeat, as these 4 keys must exist for each beat.